src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,49 +1,107 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_SIZE = 10;
+    private static final double INCREASING_COEFFICIENT = 1.5;
+
+    private T[] elementData = (T[]) new Object[DEFAULT_SIZE];
+    private int size;
 
     @Override
     public void add(T value) {
-
+        if (size == elementData.length) {
+            elementData = grow(elementData.length);
+        }
+        elementData[size] = value;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
-
+        if (index > size || index < 0) {
+            throw new ArrayListIndexOutOfBoundsException("Index passed to the method is invalid");
+        }
+        if (DEFAULT_SIZE > elementData.length || size < elementData.length) {
+            System.arraycopy(elementData, index, elementData,
+                    index + 1, size - index);
+            elementData[index] = value;
+        } else {
+            T[] widerList = grow(elementData.length);
+            System.arraycopy(elementData, index, widerList, index + 1, size - index);
+            widerList[index] = value;
+        }
+        size++;
```

Comment:
can we simplify this?
you can do the same check for as in `add(T value)` and see if the array needs to `grow()`
after this, let's just call `System.arraycopy` once

Reply:
done

Reply:
done